enterprise/server/workflow/service/service.go

```diff
@@ -565,21 +573,28 @@ func (ws *workflowService) getActions(ctx context.Context, wf *tables.Workflow,
 	}
 
 	actions := cfg.Actions
-	if actionFilter != "" {
-		actions = make([]*config.Action, 0, 1)
+	if len(actionFilter) > 0 {
+		actionMap := make(map[string]*config.Action, len(cfg.Actions))
 		for _, a := range cfg.Actions {
-			if a.Name == actionFilter {
-				actions = append(actions, a)
-				break
+			actionMap[a.Name] = a
+		}
+
+		actions = make([]*config.Action, 0, len(actionFilter))
+		for _, actionName := range actionFilter {
+			a, ok := actionMap[actionName]
+			if !ok {
+				log.Warningf("workflow action %s not found", actionName)
```

Comment:
If this is a client error (specifying an action that doesn't exist) then we shouldn't log a warning here. Maybe Info or Debug.

Also will this missing action get reported as a NotFound error in the ActionStatus in the response?

Reply:
Changed to debug 

Also refactored so that a NotFound error will be reported in the response